src/irr/asset/CImageLoaderPNG.cpp

```diff
@@ -236,18 +239,32 @@ asset::SAssetBundle CImageLoaderPng::loadAsset(io::IReadFile* _file, const asset
 	}
 
 	// Create array of pointers to rows in image data
-	RowPointers = new png_bytep[Height];
+    uint8_t** RowPointers = _IRR_NEW_ARRAY(png_bytep, Height);;
 	if (!RowPointers)
 	{
 		os::Printer::log("LOAD PNG: Internal PNG create row pointers failure\n", _file->getFileName().c_str(), ELL_ERROR);
 		png_destroy_read_struct(&png_ptr, nullptr, nullptr);
-		image->drop();
         return {};
 	}
 
+    auto texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(image->getImageDataSizeInBytes());
+    auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
+    ICPUImage::SBufferCopy& region = regions->front();
+    //region.imageSubresource.aspectMask = ...; //waits for Vulkan
+    region.imageSubresource.mipLevel = 0u;
+    region.imageSubresource.baseArrayLayer = 0u;
+    region.imageSubresource.layerCount = 1u;
+    region.bufferOffset = 0u;
+    region.bufferRowLength = 0u;//??
```

Comment:
pitch, i think in bytes

Reply:
actually not in bytes.

But you should aim for it to be image width rounded up to the next "lowest common multiple of 8 and texel block byte size" after "texel block byte size times image width"

BAsically
```cpp
uint32_t findNextAlignedBlock(uint32_t width, uint32_t blockByteSize, uint32_t _alignment=8u)
{
   for (uint32_t addr=width*blockByteSize, n=width; addr<(~0u)-blockSize+1u; width+=blockByteSize,n++)
   {
      if (addr&(_alignment-1u))
         continue;
      return n;
   }
   return 0u;
}
```

Reply:
had to change it a bit, realized this wont work